### PR TITLE
Add PSA and Notification Client docs

### PR DESCRIPTION
**Pull request by AlexFalessi**

### CODE REVIEW

File: src/docs/user/ProActiveUserGuide.adoc [hunk 1]
```diff
@@ -3044,6 +3044,131 @@ Many use cases can be fulfilled using this API. For instance:
   can scale up and down these resources when receiving the corresponding signals from users (via the Workflow Execution portal),
   or from third-party systems (via the Signals REST endpoint).
 
+== Third party services API
```
Comment (fviale) [inline src/docs/user/ProActiveUserGuide.adoc:3047]: rename to: ProActive Modules APIs

File: src/docs/user/ProActiveUserGuide.adoc [hunk 2]
```diff
@@ -3044,6 +3044,131 @@ Many use cases can be fulfilled using this API. For instance:
...
+
+Some ProActive services offers java Clients ready to be used in ProActive's workflows.
```
Comment (fviale) [inline src/docs/user/ProActiveUserGuide.adoc:3049]: In addition to the core component APIs described above, ProActive worfklow tasks can also communicate with other  ProActive Modules such as Service Automation and Notification Service.

File: src/docs/user/ProActiveUserGuide.adoc [hunk 3]
```diff
@@ -3044,6 +3044,131 @@ Many use cases can be fulfilled using this API. For instance:
...
+
+=== ProActive Service Automation (PSA) API
+
+The ProActive Service Automation API, or ProActive Service Automation Client, is a way by which a workflow can communicate with ProActive's Service Automation service
```
Comment (fviale) [inline src/docs/user/ProActiveUserGuide.adoc:3053]: ProActive's Service Automation service => the Service Automation module.
(Also link to the Service Automation documentation)

using java code to send HTTP requests. => remove this (it is not only limited to Java)

File: src/docs/user/ProActiveUserGuide.adoc [hunk 4]
```diff
@@ -3044,6 +3044,131 @@ Many use cases can be fulfilled using this API. For instance:
...
+using java code to send HTTP requests.
+
+The ProActive Service Automation Client hides the complexity of building and sending HTTP requests manually by providing
+ready-to-use classes callable from a groovy task.
```
Comment (fviale) [inline src/docs/user/ProActiveUserGuide.adoc:3057]: not only a groovy task as it's also available in some other languages (javascript, jruby, jython, probably also cpython). It's not available on many languages though (bash, cmd, R, powershell )

File: src/docs/user/ProActiveUserGuide.adoc [hunk 5]
```diff
@@ -3044,6 +3044,131 @@ Many use cases can be fulfilled using this API. For instance:
...
+[source,groovy]
+----
+// Import necessary classes from the Service Automation Client
+import org.ow2.proactive.pca.service.client.ApiClient
+import org.ow2.proactive.pca.service.client.api.ServiceInstanceRestApi
+import org.ow2.proactive.pca.service.client.model.ServiceInstanceData
+import org.ow2.proactive.pca.service.client.model.ServiceDescription
+import com.google.common.base.Strings;
+
+// Get session id
```
Comment (fviale) [inline src/docs/user/ProActiveUserGuide.adoc:3072]: // Get session id
=> 
// Get session id using the Scheduler API

File: src/docs/user/ProActiveUserGuide.adoc [hunk 6]
```diff
@@ -3044,6 +3044,131 @@ Many use cases can be fulfilled using this API. For instance:
...
+serviceInstanceData = serviceInstanceRestApi.getServiceInstance(sessionId, serviceInstanceId)
+endpoint = serviceInstanceData.getDeployments().iterator().next().getEndpoint().getUrl()
+
+// Store variables and result
+variables.put("INSTANCE_ID_" + instanceName, serviceInstanceId)
+variables.put("ENDPOINT_" + instanceName, endpoint)
+result = endpoint
+----
+
+The complete ProActive Cloud Automation Client API is available in the javadoc link:../javadoc/cloudAutomationClient[Cloud Automation Client JavaDoc].
```
Comment (fviale) [inline src/docs/user/ProActiveUserGuide.adoc:3107]: ProActive Cloud Automation Client API => ProActive Service Automation Client API
[Cloud Automation Client JavaDoc] => [Service Automation Client Javadoc]

File: src/docs/user/ProActiveUserGuide.adoc [hunk 7]
```diff
@@ -3044,6 +3044,131 @@ Many use cases can be fulfilled using this API. For instance:
...
+
+=== Notification Service API
+
+The Notification Service API, or Notification Client, is a way by which a workflow can communicate with ProActive's Notification Service
+using java code to send HTTP requests.
```
Comment (fviale) [inline src/docs/user/ProActiveUserGuide.adoc:3112]: same remark as above

File: src/docs/user/ProActiveUserGuide.adoc [hunk 8]
```diff
@@ -3044,6 +3044,131 @@ Many use cases can be fulfilled using this API. For instance:
...
+
+The Notification Client hides the complexity of building and sending HTTP requests manually by providing
+ready-to-use classes callable from a groovy task.
```
Comment (fviale) [inline src/docs/user/ProActiveUserGuide.adoc:3115]: same remark as above

File: src/docs/user/ProActiveUserGuide.adoc [hunk 9]
```diff
@@ -3044,6 +3044,131 @@ Many use cases can be fulfilled using this API. For instance:
...
+    // Send the POST request to create the Notification
+    result = eventRestApi.createEvent(sessionId, eventRequest).toString()
+    println(result)
+} catch (ApiException e) {
+    System.err.println("Exception when calling EventRestApi#createEvent")
+    e.printStackTrace();
+}
+----
+
+By using this groovy script, workflow designers can create custom Notifications anywhere in workflows.
```
Comment (fviale) [inline src/docs/user/ProActiveUserGuide.adoc:3168]: in workflows => in their workflows

File: src/docs/user/ProActiveUserGuide.adoc [hunk 10]
```diff
@@ -3044,6 +3044,131 @@ Many use cases can be fulfilled using this API. For instance:
...
+
+The complete Notification Client API is available in the javadoc link:../javadoc/notificationClient[Notification Client JavaDoc].
```
Comment (fviale) [inline src/docs/user/ProActiveUserGuide.adoc:3170]: Notification Client JavaDoc => Notification Client Javadoc

File: src/docs/user/ProActiveUserGuide.adoc (hunk 4 above)
Comment (AlexFalessi) [reply src/docs/user/ProActiveUserGuide.adoc:3057]: Should I provide an exhaustive list of the supported languages ?

File: src/docs/user/ProActiveUserGuide.adoc [hunk 11]
```diff
@@ -3044,6 +3044,129 @@ Many use cases can be fulfilled using this API. For instance:
...
+serviceInstanceData = serviceInstanceRestApi.getServiceInstance(sessionId, serviceInstanceId)
+endpoint = serviceInstanceData.getDeployments().iterator().next().getEndpoint().getUrl()
+
+// Store variables and result
+variables.put("INSTANCE_ID_" + instanceName, serviceInstanceId)
+variables.put("ENDPOINT_" + instanceName, endpoint)
+result = endpoint
+----
+
+The complete ProActive Service Automation Client API is available in the javadoc link:../javadoc/cloudAutomationClient[Service Automation Client JavaDoc].
```
Comment (fviale) [inline src/docs/user/ProActiveUserGuide.adoc:3106]: don't you need a slash at the end of the url ?
link:../javadoc/cloudAutomationClient/
or eventually:
link:../javadoc/cloudAutomationClient/index.html

File: src/docs/user/ProActiveUserGuide.adoc [hunk 12]
```diff
@@ -3044,6 +3044,129 @@ Many use cases can be fulfilled using this API. For instance:
...
+    println(result)
+} catch (ApiException e) {
+    System.err.println("Exception when calling EventRestApi#createEvent")
+    e.printStackTrace();
+}
+----
+
+By using this groovy script, workflow designers can create custom Notifications anywhere in their workflows.
+
+The complete Notification Client API is available in the javadoc link:../javadoc/notificationClient[Notification Client Javadoc].
```
Comment (fviale) [inline src/docs/user/ProActiveUserGuide.adoc:3168]: same remark as above

**Review by fviale — APPROVED** [review]
LGTM with one comment

File: src/docs/user/ProActiveUserGuide.adoc (hunk 11 above)
Comment (AlexFalessi) [reply src/docs/user/ProActiveUserGuide.adoc:3106]: Yes you are right. It seems like both a slash at the end or index.html works

**Pull request merged**